contexts/arbo-context.tsx

```diff
@@ -47,51 +47,44 @@ function filterData(data: any[], filters: { [key: string]: string[] }): any[] {
     return filterKeys.every((key: string) => {
       if (!filters[key].length) return true;
 
-      if(key === "end_date") {
-        const filterEndDate = parseISO(filters["end_date"][0]);
+      if (key === "end_date") {
+        const filterEndDate = new Date(filters["end_date"][0]);
+        const filterStartDate = new Date(filters["start_date"][0]);
 
-        if(!filterEndDate) {
-          return true;
+        if (isNaN(filterEndDate.getTime())) {
+          return true; // Handle invalid date
         }
 
-        const itemDate = new Date(item.sample_end_date);
+        const itemStartDate = new Date(item.sample_start_date);
+        const itemEndDate = new Date(item.sample_end_date);
 
-        const filterEndUTC = Date.UTC(
-          filterEndDate.getUTCFullYear(),
-          filterEndDate.getUTCMonth(),
+        // Check for any overlap in the sampling period
+        return (
+          itemEndDate <= filterEndDate ||
+          (itemEndDate >= filterEndDate && itemStartDate < filterEndDate)
         );
-
-        const itemDateUTC = Date.UTC(
-          itemDate.getUTCFullYear(),
-          itemDate.getUTCMonth(),
-        );
-        
-        return itemDateUTC <= filterEndUTC;
       }
 
-      if(key === "start_date") {
-        const filterStartDate = parseISO(filters["start_date"][0]);
+      if (key === "start_date") {
+        const filterStartDate = new Date(filters["start_date"][0]);
 
-        if(!filterStartDate) {
-          return true;
+        if (isNaN(filterStartDate.getTime())) {
+          return true; // Handle invalid date
         }
 
-        const itemDate = new Date(item.sample_start_date);
-
-        const filterStartUTC = Date.UTC(
-          filterStartDate.getUTCFullYear(),
-          filterStartDate.getUTCMonth(),
-        );
+        const itemStartDate = new Date(item.sample_start_date);
+        let itemEndDate = new Date(item.sample_end_date);
 
-        const itemDateUTC = Date.UTC(
-          itemDate.getUTCFullYear(),
-          itemDate.getUTCMonth(),
-        );
-        
+        // Check if the end date is before the start date (Fix for particular yellow fever studies in central africa that have start date 2009 and end date for 1969)
+        if (itemEndDate < itemStartDate) {
+          // Set the end date to be the same or 1 month after the start date
+          itemEndDate = new Date(itemStartDate);
+          itemEndDate.setMonth(itemEndDate.getMonth() + 1);
```

Comment:
My concern here would be instances where the item end date is in December. I would recommend something like the `addMonths` function from `date-fns` instead [docs](https://date-fns.org/v3.2.0/docs/addMonths)